patches/cryptonite-0.26.patch

```diff
@@ -0,0 +1,3536 @@
++  KeccakDigest (Object# KeccakDigest)
++  deriving Class
+ 
+-foreign import ccall "cryptonite_keccak_update"
+-    c_keccak_update :: Ptr (Context a) -> Ptr Word8 -> Word32 -> IO ()
++type instance Inherits KeccakDigest = '[JDigest]
+ 
+-foreign import ccall unsafe "cryptonite_keccak_finalize"
+-    c_keccak_finalize :: Ptr (Context a) -> Word32 -> Ptr (Digest a) -> IO ()
++c_keccak_init :: Word32 -> IO (Context a)
++c_keccak_init = fmap toContext . c_keccak_init'
++
++foreign import java unsafe "@new"
++    c_keccak_init' :: Word32 -> IO KeccakDigest
+diff --git a/Crypto/Hash/MD2.hs b/Crypto/Hash/MD2.hs
+index c878662..b21b724 100644
+--- a/Crypto/Hash/MD2.hs
++++ b/Crypto/Hash/MD2.hs
+@@ -12,6 +12,7 @@
+ {-# LANGUAGE DeriveDataTypeable #-}
+ {-# LANGUAGE DataKinds #-}
+ {-# LANGUAGE TypeFamilies #-}
++{-# LANGUAGE OverloadedStrings #-}
+ module Crypto.Hash.MD2 ( MD2 (..) ) where
+ 
+ import           Crypto.Hash.Types
+@@ -26,19 +27,6 @@ data MD2 = MD2
+ instance HashAlgorithm MD2 where
+     type HashBlockSize           MD2 = 16
+     type HashDigestSize          MD2 = 16
+-    type HashInternalContextSize MD2 = 96
+     hashBlockSize  _          = 16
+     hashDigestSize _          = 16
+-    hashInternalContextSize _ = 96
+-    hashInternalInit          = c_md2_init
+-    hashInternalUpdate        = c_md2_update
+-    hashInternalFinalize      = c_md2_finalize
+-
+-foreign import ccall unsafe "cryptonite_md2_init"
+-    c_md2_init :: Ptr (Context a)-> IO ()
+-
+-foreign import ccall "cryptonite_md2_update"
+-    c_md2_update :: Ptr (Context a) -> Ptr Word8 -> Word32 -> IO ()
+-
+-foreign import ccall unsafe "cryptonite_md2_finalize"
+-    c_md2_finalize :: Ptr (Context a) -> Ptr (Digest a) -> IO ()
++    hashInternalInit          = digest_new "MD2"
+diff --git a/Crypto/Hash/MD4.hs b/Crypto/Hash/MD4.hs
+index 0f39d58..88c4b6f 100644
+--- a/Crypto/Hash/MD4.hs
++++ b/Crypto/Hash/MD4.hs
+@@ -12,6 +12,7 @@
+ {-# LANGUAGE DeriveDataTypeable #-}
+ {-# LANGUAGE DataKinds #-}
+ {-# LANGUAGE TypeFamilies #-}
++{-# LANGUAGE OverloadedStrings #-}
+ module Crypto.Hash.MD4 ( MD4 (..) ) where
+ 
+ import           Crypto.Hash.Types
+@@ -26,19 +27,6 @@ data MD4 = MD4
+ instance HashAlgorithm MD4 where
+     type HashBlockSize           MD4 = 64
+     type HashDigestSize          MD4 = 16
+-    type HashInternalContextSize MD4 = 96
+     hashBlockSize  _          = 64
+     hashDigestSize _          = 16
+-    hashInternalContextSize _ = 96
+-    hashInternalInit          = c_md4_init
+-    hashInternalUpdate        = c_md4_update
+-    hashInternalFinalize      = c_md4_finalize
+-
+-foreign import ccall unsafe "cryptonite_md4_init"
+-    c_md4_init :: Ptr (Context a)-> IO ()
+-
+-foreign import ccall "cryptonite_md4_update"
+-    c_md4_update :: Ptr (Context a) -> Ptr Word8 -> Word32 -> IO ()
+-
+-foreign import ccall unsafe "cryptonite_md4_finalize"
+-    c_md4_finalize :: Ptr (Context
```

Comment:
Not sure if we should comment this, but extract the `Context` from the `Ptr` supposes some yak-shaving